### ISSUE TITLE
Update HTTP API not working properly when using ignoreRevs: false and a _rev in the input body

### ISSUE THREAD

**Issue by robross0606** [arangodb/arangodb#13232]
## My Environment

* __ArangoDB Version__:        Tested with 3.6.5, 3.6.6, 3.7.3 and 3.7.5
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Single Server
* __Deployment Strategy__:   Manual Start in Docker
* __Configuration__:               
* __Infrastructure__:               Docker for Desktop
* __Operating System__:        WSL2 on Windows 10
* __Total RAM in your machine__:        32Gb
* __Disks in use__:        SSD
* __Used Package__:              Docker - official Docker library

## Component, Query & Data

__Affected feature__:
HTTP API and ArangoJS driver

__Dataset__:
Any updatable document will do.

__Size of your Dataset on disk__:
N/A

__Replication Factor & Number of Shards (Cluster only)__:
N/A

## Steps to reproduce

1. Create a document
2. Use the HTTP API to update the document, setting `ignoreRevs` to `false` and placing a `_rev` value in the body of the update.

__Problem__:
As per the HTTP API documentation:

> If If-Match is not given and ignoreRevs is false and there is a _rev attribute in the body and its value does not match the revision of the document in the database, the precondition is violated.

However, this does not work.  The latest ArangoJS driver switched to using this method in 7.x line whereas it previously used the `If-Match` header.  It now appears to be broken as well (arangodb/arangojs#707).  Similarly, attempting the same call from the built-in Support->Rest API->Try It page fails in the same manner:

![MicrosoftTeams-image](https://user-images.githubusercontent.com/2965467/102277841-d87ed600-3ef6-11eb-898b-85867344fdbb.png)

The actual stored document was:
![MicrosoftTeams-image (1)](https://user-images.githubusercontent.com/2965467/102277886-e5032e80-3ef6-11eb-91c2-02d20152ba7b.png)

__Expected result__:
The HTTP API should work as documented.  If this method doesn't work, then it should be removed from the documentation and `If-Match` should be the only working method.

**Comment by robross0606** [comment]
Here are two PCAP files showing the difference between the ArangoJS 6.x and 7.x drivers.  The 6.x line uses the `If-Match` header while the 7.x line uses the second documented method of placing the `_rev` in the body.
[PCAP.zip](https://github.com/arangodb/arangodb/files/5698957/PCAP.zip)

**Comment by Simran-B** [comment]
I was able to reproduce this problem in 3.7 using the HTTP API endpoint to update a single document:

`PATCH http://localhost:8529/_api/document/<coll>/<key>`
`{ "updated": true, "_rev": "wrong_rev" }`

Returned data (successful update):

```json
{
    "_id": "<coll>/<key>",
    "_key": "<key>",
    "_rev": "_bw-oUm2---",
    "_oldRev": "_bw-oUlC--Y"
}
```

The behavior is different when bulk-updating documents:

`PATCH http://localhost:8529/_api/document/<coll>?ignoreRevs=false`
`[ { "_key": "<key>", "updated": true, "_rev": "wrong_rev" } ]`

Returned data (update rejected):

```json
[
  {
    "error": true,
    "errorNum": 1200,
    "errorMessage": "conflict"
  }
]
```

I need to check with engineering whether it's intentional that only the `If-Match` header works as precondition for single document updates. The paragraph about `ignoreRevs` and `_rev` in the body might belong to the bulk update description only.

**Comment by Simran-B** [comment]
We finally found the root cause of the problem.

There are two options when updating a single document with revision check:
1. `If-Match` header
2. `?ignoreRevs=false` and `_rev` in the body

The target document is specified in the URL path, but in the latter case it is somehow necessary to specify the document key in the body again or a `412 Precondition failed` error is thrown:

`PATCH /_api/document/<coll>/<key>?ignoreRevs=false`
`{ "_rev": "<rev>" }` - fails

`PATCH /_api/document/<coll>/<key>?ignoreRevs=false`
`{ "_key": "<key>", "_rev": "<rev>" }` - succeeds

I suppose the intention is to make updates fail if the payload `_key` isn't matching the key in the URL path, but if no `_key` attribute is present in the body, then it's not quite clear whether that should count as a mismatch or not (the implicit value of non-existing attributes is null in AQL).

One of our engineers confirmed that there's a bug in the code when handling this case however, and that the request should succeed without having to specify the `_key` in the body.

```diff
         if (headerRev.isSet()) {
           builder.add(StaticStrings::RevString, VPackValue(headerRev.toString()));
         } else if (!opOptions.ignoreRevs && revInBody.isSet()) {
-          builder.add(StaticStrings::RevString, VPackValue(headerRev.toString()));
+          builder.add(StaticStrings::RevString, VPackValue(revInBody.toString()));
        }
```

`headerRev` is mistakenly used in the latter branch, but actually empty. The revision from the body is supposed to be taken. A fix for this is coming up.

**Comment by Simran-B** [comment]
With the fixes released in 3.6.12 and 3.7.7, the following request succeeds now:
`PATCH /_api/document/<coll>/<key>?ignoreRevs=false`
`{ "_rev": "<rev>" }` (no `_key` supplied here)
